src/main/scala/edu/ie3/simona/service/primary/PrimaryServiceWorker.scala

```diff
@@ -362,25 +382,35 @@ object PrimaryServiceWorker extends SimonaService {
     *
     * @param baseStateData
     *   The base state data to update
+    * @param currentTick
+    *   The current tick for which data was provided.
     * @return
     *   Updated base state data and an option on a sequence of schedule trigger
     *   messages
     */
   private def updateStateDataAndBuildTriggerMessages[V <: Value](
-      baseStateData: PrimaryServiceInitializedStateData[V]
+      baseStateData: PrimaryServiceInitializedStateData[V],
+      currentTick: Long,
   ): (
       PrimaryServiceInitializedStateData[V],
       Option[Long],
   ) = {
-    val (maybeNextActivationTick, remainderActivationTicks) =
-      baseStateData.activationTicks.pop
-    (
-      baseStateData.copy(
-        maybeNextActivationTick = maybeNextActivationTick,
-        activationTicks = remainderActivationTicks,
-      ),
-      maybeNextActivationTick,
-    )
+    val ticks = baseStateData.activationTicks
+
+    baseStateData.maybeNextActivationTick match {
+      case Some(nextTick) if nextTick > currentTick =>
```

Comment:
This does not happen at all, I think. The previous code was fine the way it was. Or do you have any counter-example?

Reply:
There is a test, where we request data for tick 200 an receive a completion for tick 900. Therefore, I added this code. Otherwise we would not be scheduled for tick 900.

If you have a better solution for the problem, please let me know.

Reply:
I find it interesting that the actual tick value is not checked. Maybe we should add some code to check the value of the activation message. Or is this already handled elsewhere.